starcheck/calc_ccd_temps.py

```diff
@@ -152,8 +152,19 @@ def get_ccd_temps(oflsdir, outdir='out',
 
     # Get tstart, tstop, commands from backstop file in opt.oflsdir
     bs_cmds = get_bs_cmds(oflsdir)
-    tstart = DateTime(bs_cmds[0]['date']).secs
-    tstop = DateTime(bs_cmds[-1]['date']).secs
+
+    # Use RLTT and SCHEDULED_STOP_TIME if available
+    rltt = bs_cmds['event_type'] == 'RUNNING_LOAD_TERMINATION_TIME'
+    if np.any(rltt):
+        tstart = DateTime(bs_cmds['date'][rltt][0]).secs
```

Comment:
This needs to be `tstart = DateTime(bs_cmds['date'][rltt][0]).secs + 0.001`.  The RLTT is defined so that running commands at exactly the RLTT are *included* in propagation.  However `get_cmds()` uses the convention of getting commands `date_start <= cmd_date < date_stop`.  (All done using standard date strings to 1 millisec precision).

https://cxc.cfa.harvard.edu/mta/ASPECT/tool_doc/kadi/api.html#kadi.commands.commands.get_cmds

Reply:
Gotcha.  I still hadn't understood the plan but figured just tossing it in here would help.  I'd be helped more by some worked examples in test cases.

Reply:
The recent interrupt schedule had an example of this.  But maybe you are overthinking this since it is really no more than "chop running commands after RLTT, propagate out to scheduled stop".  That's it.  

All the words John sent about how MP chooses the times are not relevant except for FOT MP is guaranteeing that "chop running commands after RLTT, propagate out to scheduled stop" will give the correct set of commands for propagation.

Reply:
"The recent interrupt schedule had an example of this."  I thought the schedule was incorrect about RLTT and they were going to change for future use.  And I didn't have the "right answer".  But that's fine.

Reply:
Indeed the RLTT for APR1420 would include an AOACRSTD command that never ran.  Though from the perspective of "giving the correct set of commands for propagation", it was just fine because AOACRSTD doesn't affect propagation.  In fact that more functional definition (giving correct propagation) is really applicable because in the case of an interrupt there might be transmitter off commands before the RLTT that never actually get run, but we don't care.

Reply:
Also, why did the AOACRSTD not run?  I thought the RLTT in the APR1420 backstop was really describing the time to terminate the APR1320 commands running onboard.  The AOACRSTD command was in the APR1420 backstop.  Does it only run contingent upon something else?

Reply:
I was referring to the AOACRSTD at 2020:105:01:05:22.340 in APR1320 that was being interrupted since we have "Estimated time of on-board load stop time: 2020:104:22:25:00.000" (referring to the real-time activity of stopping loads).

The way I said it was admittedly ambiguous in language.  How about "Indeed the RLTT specified in the APR1420 loads would specify to include (in the propagation commands) an AOACRSTD command from APR1320A that occurs well after the actual interrupt time during comm."  But this is because of a one-time process mistake and an off-by-1msec error.  Going forward that won't happen.

Takeaway: the RLTT is not *exactly* related to when loads will actually be interrupted by operators during comm.  It will normally be after the comm, with the only hard requirement that chopping commands using the RLTT will give the correct propagation.

Reply:
Ah.  I didn't realize that there were *two* commands at exactly the RLTT for this interrupt, the AOACRCSTD at 2020:105:01:05:22.340 in the interrupted and interrupting schedules.  I assume the AOACRCSTD from APR1420A did run (and it doesn't really matter).  Your comment here about RLTT not necessarily being when the loads will actually be interrupted is also helpful to me (not for this starcheck exercise but in general).